app/invocation/invocation_action_card.tsx

```diff
@@ -196,6 +225,47 @@ export default class InvocationActionCardComponent extends React.Component<Props
     );
   }
 
+  handleFileClicked(node: any) {
+    let digest_string = node.digest.hash + "/" + node.digest.sizeBytes;
+    let dirFile = "bytestream://" + this.getCacheAddress() + "/blobs/" + digest_string;
+    if (!node.hasOwnProperty("isExecutable")) {
+      if (this.state.treeShaToExpanded.get(digest_string)) {
+        this.state.treeShaToExpanded.set(digest_string, false);
+        this.setState({ treeShaToExpanded: this.state.treeShaToExpanded });
+        return;
+      }
+      rpcService
+        .fetchBytestreamFile(dirFile, this.props.model.getId(), "arraybuffer")
+        .then((dir_buff: any) => {
+          let tempDir = build.bazel.remote.execution.v2.Directory.decode(new Uint8Array(dir_buff));
+          this.state.treeShaToExpanded.set(digest_string, true);
+          let contents: (any | any)[] = [];
+          this.state.treeShaToChildrenMap.set(
+            digest_string,
+            contents.concat(tempDir.directories).concat(tempDir.files)
+          );
+          this.setState({
+            treeShaToChildrenMap: this.state.treeShaToChildrenMap,
+            treeShaToExpanded: this.state.treeShaToExpanded,
+          });
+        })
+        .catch(() => {
+          console.error("Error loading bytestream file profile!");
+          this.setState({
+            ...this.state,
+            error: "Error loading action profile. Make sure your cache is correctly configured.",
+          });
+        });
+      return;
+    } else {
```

Comment:
this would probably be less complex if you did an early exit instead of an if/else 

```
if (node.hasOwnProperty("isExecutable")) {
      try {
        rpcService.downloadBytestreamFile(node.name, dirFile, this.props.model.getId());
      } catch {
        console.error("Error downloading bytestream timing profile");
      }
}

The rest of the code above ...
```